src/main/java/org/generationcp/middleware/api/germplasm/GermplasmService.java

```diff
@@ -112,4 +114,9 @@
 
 	long countAttributesByGUID(String gemrplasmUUID, List<String> attributeDbIds);
 
+	List<GermplasmNameTypeDTO> filterGermplasmNameTypes(Set<String> codes);
```

Comment:
It doesn't make intuitive sense to me to offer this as part of `GermplasmService` as it's not action to a specific (or group of) actual germplasm.

Reply:
We have the same problem with attributes but we know there should be a refactor. As of now I would not mind that much on where we locate this function. It was previously in BMSAPI calling 
`germplasmService.getUserDefinedFieldByTableTypeAndCodes `
which was also weird.
I prefer to wait a bit to check what stays as part of user defined fields and what will be moved to ontology to make a decision on where we should place all this kind of metadata code